### ISSUE TITLE
Error when starting weewxd

### ISSUE THREAD

**Issue by mjm455** [weewx/weewx#963]
I ran "sudo weectl station reconfigure" and picked Simulator but the error is mostly similar:

Jun 7 04:41:21 raspberrypi systemd[1]: Started WeeWX.
Jun 7 04:41:22 raspberrypi weewxd[17162]: INFO main: Initializing weewxd version 5.0.2
Jun 7 04:41:22 raspberrypi weewxd[17162]: INFO main: Command line: /usr/share/weewx/weewxd.py /etc/weewx/weewx.conf
Jun 7 04:41:22 raspberrypi weewxd[17162]: INFO main: Using Python 3.9.2 (default, Mar 12 2021, 04:06:34) https://github.com/themoosman/weewx-ambientweatherapi-json/issues/12[GCC 10.2.1 20210110]
Jun 7 04:41:22 raspberrypi weewxd[17162]: INFO main: Located at /usr/bin/python3
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: Platform Linux-6.1.21+-armv6l-with-glibc2.31
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: Locale: 'en_GB.UTF-8'
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: Entry path: /usr/share/weewx/weewxd.py
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: WEEWX_ROOT: /etc/weewx
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: Configuration file: /etc/weewx/weewx.conf
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: User module: /etc/weewx/bin/user
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO main: Debug: 0
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.engine: Loading station type Simulator (weewx.drivers.simulator)
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.engine: StdConvert target unit is 0x1
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.wxservices: StdWXCalculate will use data binding wx_binding
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.engine: Archive will use data binding wx_binding
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.engine: Record generation will be attempted in 'hardware'
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.engine: Using archive interval of 300 seconds (specified in weewx configuration)
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.restx: StationRegistry: Station will be registered.
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.restx: Wunderground: Posting not enabled.
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.restx: PWSweather: Posting not enabled.
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.restx: CWOP: Posting not enabled.
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.restx: WOW: Posting not enabled.
Jun 7 04:41:23 raspberrypi weewxd[17162]: INFO weewx.restx: AWEKAS: Posting not enabled.
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: Caught unrecoverable exception:
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** init() takes 1 positional argument but 3 were given
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** Traceback (most recent call last):
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** File "/usr/share/weewx/weewxd.py", line 160, in main
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** engine = weewx.engine.StdEngine(config_dict)
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** File "/usr/share/weewx/weewx/engine.py", line 89, in init
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** self.loadServices(config_dict)
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** File "/usr/share/weewx/weewx/engine.py", line 157, in loadServices
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** obj = weeutil.weeutil.get_object(svc)(self, config_dict)
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** TypeError: init() takes 1 positional argument but 3 were given
Jun 7 04:41:24 raspberrypi weewxd[17162]: CRITICAL main: **** Exiting.
Jun 7 04:41:24 raspberrypi weewxd[17162]: Traceback (most recent call last):
Jun 7 04:41:24 raspberrypi weewxd[17162]: File "/usr/share/weewx/weewxd.py", line 265, in
Jun 7 04:41:24 raspberrypi weewxd[17162]: main()
Jun 7 04:41:24 raspberrypi weewxd[17162]: File "/usr/share/weewx/weewxd.py", line 160, in main
Jun 7 04:41:24 raspberrypi weewxd[17162]: engine = weewx.engine.StdEngine(config_dict)
Jun 7 04:41:24 raspberrypi weewxd[17162]: File "/usr/share/weewx/weewx/engine.py", line 89, in init
Jun 7 04:41:24 raspberrypi weewxd[17162]: self.loadServices(config_dict)
Jun 7 04:41:24 raspberrypi weewxd[17162]: File "/usr/share/weewx/weewx/engine.py", line 157, in loadServices
Jun 7 04:41:24 raspberrypi weewxd[17162]: obj = weeutil.weeutil.get_object(svc)(self, config_dict)
Jun 7 04:41:24 raspberrypi weewxd[17162]: TypeError: init() takes 1 positional argument but 3 were given
Jun 7 04:41:25 raspberrypi systemd[1]: weewx.service: Main process exited, code=exited, status=1/FAILURE
Jun 7 04:41:25 raspberrypi systemd[1]: weewx.service: Failed with result 'exit-code'.
Jun 7 04:41:25 raspberrypi systemd[1]: weewx.service: Consumed 3.927s CPU time.

**Comment by tkeffer** [comment]
Please run the utility `weectl debug --output=/var/tmp/weewx.txt`. Look over the file `/var/tmp/weewx.tmp` to make sure that all of your passwords have been obfuscated, then post the file. 

It will contain a copy of your configuration file `weewx.conf`. I suspect one of the options got malformed.

**Comment by mjm455** [comment]
Thanks for your help on this.  I was about to refresh the whole OS and start over but I haven't really done much to it.  I'm installing this on a Raspberry Pi Zero W.  Here is the output:

pi@raspberrypi:/etc/weewx $ more /var/tmp/weewx.txt

System info
  Platform:       Linux-6.1.21+-armv6l-with-glibc2.31
  Python Version: 3.9.2

Load Information
   1 minute load average:  0.08
   5 minute load average:  0.03
  15 minute load average:  0.01

General Weewx info
  Weewx version 5.0.2 detected.

Currently installed extensions
Extension Name    Version   Description
ambientweatherapi 0.0.11    WeeWx AmbientWeather API Driver.
interceptor       0.60      Capture weather data from HTTP requests

Archive info
  Database name:        weewx.sdb
  Table name:           archive
  Version               4.0
  Unit system:          None (Unknown unit constant)
  First good timestamp: ******* N/A *******     (    N/A   )
  Last good timestamp:  ******* N/A *******     (    N/A   )
  (no archive records found)
                        It is likely that the database (weewx.sdb) archive table (archive)
                        exists but contains no data.
  weewx (weewx.conf) is set to use an archive interval of 300 seconds.
  The station hardware was not interrogated to determine the archive interval.

Supported SQL keys
  dateTime                  usUnits                   interval
  altimeter                 appTemp                   appTemp1
  barometer                 batteryStatus1            batteryStatus2
  batteryStatus3            batteryStatus4            batteryStatus5
  batteryStatus6            batteryStatus7            batteryStatus8
  cloudbase                 co                        co2
  consBatteryVoltage        dewpoint                  dewpoint1
  ET                        extraHumid1               extraHumid2
  extraHumid3               extraHumid4               extraHumid5
  extraHumid6               extraHumid7               extraHumid8
  extraTemp1                extraTemp2                extraTemp3
  extraTemp4                extraTemp5                extraTemp6
  extraTemp7                extraTemp8                forecast
  hail                      hailBatteryStatus         hailRate
  heatindex                 heatindex1                heatingTemp
  heatingVoltage            humidex                   humidex1
  inDewpoint                inHumidity                inTemp
  inTempBatteryStatus       leafTemp1                 leafTemp2
  leafWet1                  leafWet2                  lightning_distance
  lightning_disturber_count lightning_energy          lightning_noise_count
  lightning_strike_count    luminosity                maxSolarRad
  nh3                       no2                       noise
  o3                        outHumidity               outTemp
  outTempBatteryStatus      pb                        pm10_0
  pm1_0                     pm2_5                     pressure
  radiation                 rain                      rainBatteryStatus
  rainRate                  referenceVoltage          rxCheckPercent
  signal1                   signal2                   signal3
  signal4                   signal5                   signal6
  signal7                   signal8                   snow
  snowBatteryStatus         snowDepth                 snowMoisture
  snowRate                  so2                       soilMoist1
  soilMoist2                soilMoist3                soilMoist4
  soilTemp1                 soilTemp2                 soilTemp3
  soilTemp4                 supplyVoltage             txBatteryStatus
  UV                        uvBatteryStatus           windBatteryStatus
  windchill                 windDir                   windGust
  windGustDir               windrun                   windSpeed

Databases configured in weewx.conf:
  archive_sqlite:
     database_name weewx.sdb
            driver weedb.sqlite
       SQLITE_ROOT /var/lib/weewx
  archive_mysql:
     database_name weewx
            driver weedb.mysql
              host localhost
              user weewx
          password weewx

--- Start configuration file /etc/weewx/weewx.conf ---
# WEEWX CONFIGURATION FILE
#
# Copyright (c) 2009-2024 Tom Keffer <tkeffer@gmail.com>
# See the file LICENSE.txt for your rights.

##############################################################################

# This section is for general configuration information.

# Set to 1 for extra debug info, otherwise comment it out or set to zero.
debug = 0

# Whether to log successful operations. May get overridden below.
log_success = True

# Whether to log unsuccessful operations. May get overridden below.
log_failure = True

# This configuration file was created by ...
version = 5.0.2
config_path = /etc/weewx/weewx.conf
WEEWX_ROOT = /etc/weewx

##############################################################################

#   This section is for information about the station.

[Station]

    # Description of the station location, such as your town.
    location = "XXXXXXXXXXX"

    # Latitude in decimal degrees. Negative for southern hemisphere.
    latitude = Xx.045
    # Longitude in decimal degrees. Negative for western hemisphere.
    longitude = -Yy.097

    # Altitude of the station, with the unit it is in. This is used only
    # if the hardware cannot supply a value.
    altitude = XXx, foot    # Choose 'foot' or 'meter' for unit

    # Set to type of station hardware. There must be a corresponding stanza
    # in this file, which includes a value for the 'driver' option.
    station_type = Simulator

    # If you have a website, you may specify an URL
    station_url = XXXXXX

    # The start of the rain year (1=January; 10=October, etc.). This is
    # downloaded from the station if the hardware supports it.
    rain_year_start = 1

    # Start of week (0=Monday, 6=Sunday)
    week_start = 6

##############################################################################

[Simulator]
    # This section is for the weewx weather station simulator.

    # The time (in seconds) between LOOP packets.
    loop_interval = 2.5

    # The simulator mode can be either 'simulator' or 'generator'.
    # Real-time simulator. Sleep between each LOOP packet.
    mode = simulator
    # Generator.  Emit LOOP packets as fast as possible (useful for testing).
    #mode = generator

    # The start time. Format is YYYY-mm-ddTHH:MM. If not specified, the
    # default is to use the present time.
    #start = 2011-01-01T00:00

    # The driver to use.
    driver = weewx.drivers.simulator

##############################################################################

[WS28xx]
    # This section is for the La Crosse WS-2800 series of weather stations.

    # Radio frequency to use between USB transceiver and console: US or EU
    # US uses 915 MHz, EU uses 868.3 MHz.  Default is US.
    transceiver_frequency = US

    # The station model, e.g., 'LaCrosse C86234' or 'TFA Primus'
    model = Ambient Weather WS-2902

    # The driver to use:
    driver = weewx.drivers.ws28xx

##############################################################################


[ambientweatherapi]
    # This section for the AmbientWeatheAPI driver for weewx

    # The time (in seconds) between LOOP packets and getting API data
    # The ObserverIP only pushes data every 60 seconds.
    loop_interval = 60

    #API Log level
    log_level = ERROR

    #Log file location
    log_file = /var/log/weewx.log

    #URL to the Ambient Weather API
    api_url = https://api.ambientweather.net/v1

    #Ambient Weather API App Key
    api_app_key = YYYYy

    #Ambient Weather API Key
    api_key = XXXXXX

    #MAC Address of station to reutrn data (default is the first station in the list.)
    #station_mac = 'E8:DB:84:E3:DA:CC'

    #Ambient Weather Use Meteobridge (default if False)
    #use_meteobridge = ''

    # Name of Hardware device.
    hardware = Ambient Weather WS-2902

    # The driver to use:
    driver = user.ambientweatherapi

#[ambientweatherapi]
#    driver = user.ambientweatherapi
#    loop_interval = 120
#    api_url = https://api.ambientweather.net/v1
#    api_app_key = XXXXXx
#    api_key = YYYYYYYY
#    hardware = My Weather Station

##############################################################################

[Interceptor]
    # This section is for the network traffic interceptor driver.

    # The driver to use:
    driver = user.interceptor

    # Specify the hardware device to capture. Options include:
    #   acurite-bridge - acurite internet bridge, smarthub, or access
    #   observer - fine offset WH2600/HP1000/HP1003, ambient WS2902
    #   lw30x - oregon scientific LW301/LW302
    #   lacrosse-bridge - lacrosse GW1000U/C84612 internet bridge
    #   ecowitt-client - any hardware that uses the ecowitt protocol
    #   wu-client - any hardware that uses the weather underground protocol
    device_type = acurite-bridge

##############################################################################

#   This section is for uploading data to Internet sites

[StdRESTful]

    # Uncomment and change to override logging for uploading services.
    # log_success = True
    # log_failure = True

    [[StationRegistry]]
        # To register this weather station at weewx.com, set this to true, and
        # set option 'station_url', located in the [Station] section above.
        register_this_station = True

    [[AWEKAS]]
        # This section is for configuring posts to AWEKAS.

        # If you wish to post to AWEKAS, set the option 'enable' to true, then
        # specify a username and password.
        # Use quotes around the password to guard against parsing errors.
        enable = false
        username = XXXXXX
        password = XXXXXX

    [[CWOP]]
        # This section is for configuring posts to CWOP.

        # If you wish to post to CWOP, set the option 'enable' to true,
        # then specify the station ID (e.g., CW1234).
        enable = false
        station = XXXXXX
        # If this is an APRS (radio amateur) station, specify the
        # passcode (e.g., 12345). Otherwise, ignore.
        passcode = XXXXXX

    [[PWSweather]]
        # This section is for configuring posts to PWSweather.com.

        # If you wish to post to PWSweather.com, set the option 'enable' to
        # true, then specify a station and password.
        # Use quotes around the password to guard against parsing errors.
        enable = false
        station = XXXXXX
        password = XXXXXX

    [[WOW]]
        # This section is for configuring posts to WOW.

        # If you wish to post to WOW, set the option 'enable' to true, then
        # specify a station and password.
        # Use quotes around the password to guard against parsing errors.
        enable = false
        station = XXXXXX
        password = XXXXXX

    [[Wunderground]]
        # This section is for configuring posts to the Weather Underground.

        # If you wish to post to the Weather Underground, set the option
        # 'enable' to true,  then specify a station (e.g., 'KORHOODR3'). Use
        # the station key (find it at
        # https://www.wunderground.com/member/devices) for the password.
        enable = false
        station = XXXXXX
        password = XXXXXX

        # Set the following to True to have weewx use the WU "Rapidfire"
        # protocol. Not all hardware can support it. See the User's Guide.
        rapidfire = False

##############################################################################

#   This section specifies what reports, using which skins, to generate.

[StdReport]

    # Where the skins reside, relative to WEEWX_ROOT
    SKIN_ROOT = skins

    # Where the generated reports should go, relative to WEEWX_ROOT
    HTML_ROOT = /var/www/html/weewx

    # Uncomment and change to override logging for reports.
    # log_success = True
    # log_failure = True

    # The database binding indicates which data should be used in reports.
    data_binding = wx_binding

    # Each of the following subsections defines a report that will be run.
    # See the customizing guide to change the units, plot types and line
    # colors, modify the fonts, display additional sensor data, and other
    # customizations. Many of those changes can be made here by overriding
    # parameters, or by modifying templates within the skin itself.

    [[SeasonsReport]]
        # The SeasonsReport uses the 'Seasons' skin, which contains the
        # images, templates and plots for the report.
        skin = Seasons
        enable = true

    [[SmartphoneReport]]
        # The SmartphoneReport uses the 'Smartphone' skin, and the images and
        # files are placed in a dedicated subdirectory.
        skin = Smartphone
        enable = false
        HTML_ROOT = /var/www/html/weewx/smartphone

    [[MobileReport]]
        # The MobileReport uses the 'Mobile' skin, and the images and files
        # are placed in a dedicated subdirectory.
        skin = Mobile
        enable = false
        HTML_ROOT = /var/www/html/weewx/mobile

    [[StandardReport]]
        # This is the old "Standard" skin. By default, it is not enabled.
        skin = Standard
        enable = false

    [[FTP]]
        # FTP'ing the results to a webserver is treated as just another report,
        # albeit one with an unusual report generator!
        skin = Ftp

        # If you wish to use FTP, set "enable" to "true", then fill out the
        # next four lines.
        # Use quotes around the password to guard against parsing errors.
        enable = false
        user = XXXXXX
        password = XXXXXX
        server = XXXXXX    # The ftp server name, e.g, www.myserver.org
        path = replace_me    # The destination directory, e.g., /weather

        # Set to True for an FTP over TLS (FTPS) connection. Not all servers
        # support this.
        secure_ftp = False

        # To upload files from something other than what HTML_ROOT is set
        # to above, specify a different HTML_ROOT here.
        #HTML_ROOT = /var/www/html/weewx

        # Most FTP servers use port 21.
        port = 21

        # Set to 1 to use passive mode, zero for active mode
        passive = 1

    [[RSYNC]]
        # rsync'ing to a webserver is treated as just another report.
        skin = Rsync

        # If you wish to use rsync, you must configure passwordless ssh using
        # public/private key authentication from the user account that weewx
        # runs to the user account on the remote machine where the files
        # will be copied.
        #
        # If you wish to use rsync, set "enable" to "true", then
        # fill out server, user, and path.
        # The server should appear in your .ssh/config file.
        # The user is the username used in the identity file.
        # The path is the destination directory, such as /var/www/html/weather.
        # Be sure that the user has write permissions on the destination!
        enable = false
        server = XXXXXX
        user = XXXXXX
        path = replace_me

        # To upload files from something other than what HTML_ROOT is set
        # to above, specify a different HTML_ROOT here.
        #HTML_ROOT = /var/www/html/weewx

        # Rsync can be configured to remove files from the remote server if
        # they don't exist under HTML_ROOT locally. USE WITH CAUTION: if you
        # make a mistake in the remote path, you could could unintentionally
        # cause unrelated files to be deleted. Set to 1 to enable remote file
        # deletion, zero to allow files to accumulate remotely.
        delete = 0

    # Options in the [[Defaults]] section below will apply to all reports.
    # What follows are a few of the more popular options you may want to
    # uncomment, then change.
    [[Defaults]]

        # Which language to use for all reports. Not all skins support all
        # languages.  You can override this for individual reports.
        lang = en

        # Which unit system to use for all reports. Choices are 'us', 'metric',
        # or 'metricwx'.  You can override this for individual reports.
        unit_system = us

        [[[Units]]]

            # Option "unit_system" above sets the general unit system, but
            # overriding specific unit groups is possible. These are popular
            # choices. Uncomment and set as appropriate.  The unit is always
            # in the singular, e.g., 'mile_per_hour', NOT 'miles_per_hour'
            [[[[Groups]]]]
                # group_altitude     = meter              # Options are 'foot' or 'meter'
                # group_pressure     = mbar               # Options are 'inHg', 'mmHg', 'mbar', or 'hPa'
                # group_rain         = mm                 # Options are 'inch', 'cm', or 'mm'
                # group_rainrate     = mm_per_hour        # Options are 'inch_per_hour', 'cm_per_hour', or 'mm_per_hour'
                # group_temperature  = degree_C           # Options are 'degree_C', 'degree_F', or 'degree_K'
                # The following line is used to keep the above lines indented
                # properly. It can be ignored.
                unused = unused

            # Uncommenting the following section frequently results in more
            # attractive formatting of times and dates, but may not work in
            # your locale.
            [[[[TimeFormats]]]]
                # day        = %H:%M
                # week       = %H:%M on %A
                # month      = %d-%b-%Y %H:%M
                # year       = %d-%b-%Y %H:%M
                # rainyear   = %d-%b-%Y %H:%M
                # current    = %d-%b-%Y %H:%M
                # ephem_day  = %H:%M
                # ephem_year = %d-%b-%Y %H:%M
                # The following line is used to keep the above lines indented
                # properly. It can be ignored.
                unused = unused

        [[[Labels]]]
            # Users frequently change the labels for these observation types.
            [[[[Generic]]]]
                # inHumidity     = Inside Humidity
                # inTemp         = Inside Temperature
                # outHumidity    = Outside Humidity
                # outTemp        = Outside Temperature
                # extraTemp1     = Temperature1
                # extraTemp2     = Temperature2
                # extraTemp3     = Temperature3
                # The following line is used to keep the above lines indented
                # properly. It can be ignored.
                unused = unused

##############################################################################

#   This service converts the unit system coming from the hardware to a unit
#   system in the database.

[StdConvert]

    # The target_unit affects only the unit system in the database. Once
    # chosen it cannot be changed without converting the entire database.
    # Modification of target_unit after starting weewx will result in
    # corrupt data - the database will contain a mix of US and METRIC data.
    #
    # The value of target_unit does not affect the unit system for
    # reporting - reports can display US, Metric, or any combination of units.
    #
    # In most cases, target_unit should be left as the default: US
    #
    # In particular, those migrating from a standard wview installation
    # should use US since that is what the wview database contains.

    # DO NOT MODIFY THIS VALUE UNLESS YOU KNOW WHAT YOU ARE DOING!
    target_unit = US    # Options are 'US', 'METRICWX', or 'METRIC'

##############################################################################

#   This section can adjust data using calibration expressions.

[StdCalibrate]

    [[Corrections]]
        # For each type, an arbitrary calibration expression can be given.
        # It should be in the units defined in the StdConvert section.
        # Example:
        foo = foo + 0.2

##############################################################################

#   This section is for quality control checks. If units are not specified,
#   values must be in the units defined in the StdConvert section.

[StdQC]

    [[MinMax]]
        barometer = 26, 32.5, inHg
        pressure = 24, 34.5, inHg
        outTemp = -40, 120, degree_F
        inTemp = 10, 120, degree_F
        outHumidity = 0, 100
        inHumidity = 0, 100
        windSpeed = 0, 120, mile_per_hour
        rain = 0, 10, inch

##############################################################################

#   This section controls the origin of derived values.

[StdWXCalculate]

    [[Calculations]]
        # How to calculate derived quantities.  Possible values are:
        #  hardware        - use the value provided by hardware
        #  software        - use the value calculated by weewx
        #  prefer_hardware - use value provide by hardware if available,
        #                      otherwise use value calculated by weewx

        pressure = prefer_hardware
        altimeter = prefer_hardware
        appTemp = prefer_hardware
        barometer = prefer_hardware
        cloudbase = prefer_hardware
        dewpoint = prefer_hardware
        ET = prefer_hardware
        heatindex = prefer_hardware
        humidex = prefer_hardware
        inDewpoint = prefer_hardware
        maxSolarRad = prefer_hardware
        rainRate = prefer_hardware
        windchill = prefer_hardware
        windrun = prefer_hardware

##############################################################################

#   For hardware that supports it, this section controls how often the
#   onboard clock gets updated.

[StdTimeSynch]

    # How often to check the weather station clock for drift (in seconds)
    clock_check = 14400

    # How much it can drift before we will correct it (in seconds)
    max_drift = 5

##############################################################################

#   This section is for configuring the archive service.

[StdArchive]

    # If the station hardware supports data logging then the archive interval
    # will be downloaded from the station. Otherwise, specify it (in seconds).
    archive_interval = 300

    # If possible, new archive records are downloaded from the station
    # hardware. If the hardware does not support this, then new archive
    # records will be generated in software.
    # Set the following to "software" to force software record generation.
    record_generation = hardware

    # Whether to include LOOP data in hi/low statistics.
    loop_hilo = True

    # Uncomment and change to override logging for archive operations.
    # log_success = True
    # log_failure = True

    # The data binding used to save archive records.
    data_binding = wx_binding

##############################################################################

#   This section binds a data store to a database.

[DataBindings]

    [[wx_binding]]
        # The database must match one of the sections in [Databases].
        # This is likely to be the only option you would want to change.
        database = archive_sqlite
        # The name of the table within the database.
        table_name = archive
        # The manager handles aggregation of data for historical summaries.
        manager = weewx.manager.DaySummaryManager
        # The schema defines the structure of the database.
        # It is *only* used when the database is created.
        schema = schemas.wview_extended.schema

##############################################################################

#   This section defines various databases.

[Databases]

    # A SQLite database is simply a single file.
    [[archive_sqlite]]
        database_name = weewx.sdb
        database_type = SQLite

    # MySQL
    [[archive_mysql]]
        database_name = weewx
        database_type = MySQL

##############################################################################

#   This section defines defaults for the different types of databases.

[DatabaseTypes]

    # Defaults for SQLite databases.
    [[SQLite]]
        driver = weedb.sqlite
        # Directory in which database files are located, relative to WEEWX_ROOT
        SQLITE_ROOT = /var/lib/weewx

    # Defaults for MySQL databases.
    [[MySQL]]
        driver = weedb.mysql
        # The host where the database is located.
        host = localhost
        # The user name for logging in to the host.
        user = XXXXXX
        # Use quotes around the password to guard against parsing errors.
        password = XXXXXX

##############################################################################

#   This section configures the internal weewx engine.

[Engine]

    # This section specifies which services should be run and in what order.
    [[Services]]
        prep_services = weewx.engine.StdTimeSynch
        data_services = ,
        process_services = weewx.engine.StdConvert, weewx.engine.StdCalibrate, weewx.engine.StdQC, weewx.wxservices.StdWXCalculate
        xtype_services = weewx.wxxtypes.StdWXXTypes, weewx.wxxtypes.StdPressureCooker, weewx.wxxtypes.StdRainRater, weewx.wxxtypes.StdDelta
        archive_services = weewx.engine.StdArchive
        restful_services = weewx.restx.StdStationRegistry, weewx.restx.StdWunderground, weewx.restx.StdPWSweather, weewx.restx.StdCWOP, weewx.restx.StdWOW, weewx.restx
.StdAWEKAS, user.ambientweatherapi.AmbientWeatherAPI
        report_services = weewx.engine.StdPrint, weewx.engine.StdReport

**Comment by tkeffer** [comment]
Best I can tell, the extension `ambientweatherapi` is not initializing the driver correctly. It is doing

```
def loader(config_dict, engine):
    station = AmbientWeatherAPI(**config_dict[DRIVER_NAME])
    return station
```

but it should be 

```
def loader(config_dict, engine):
    station = AmbientWeatherAPI(engine, config_dict))
    return station
```

See [line 26](https://github.com/themoosman/weewx-ambientweatherapi-json/blob/e4637ddac9a968170a032a2ec76bae2a85570e53/bin/user/ambientweatherapi.py#L24C1-L27C19).

**Comment by mjm455** [comment]
But the base install is not working for me so it cannot be the extension.  Do I need to refresh the OS and start from scratch?  Not sure what else to try.

**Comment by tkeffer** [comment]
Not sure what you mean by "the base install is not working." Do you mean the installation of WeeWX? From your first post, it looks like WeeWX installed just fine, but the extension is not working. 

If WeeWX installed properly, it should not be necessary to reinstall the OS or WeeWX. Just fix the installation and reinstall it.

Or, am I misinterpreting what I'm seeing?

For the moment, I'll reopen the issue.

**Comment by tkeffer** [comment]
I'm afraid that I may have led you slightly astray. 

I tried installing the ambient weather extension and got the same error as you. The problem is that it tries to install itself as a WeeWX service. That is, it includes itself among the RESTful services in option `restful_services`. See [line 17](https://github.com/themoosman/weewx-ambientweatherapi-json/blob/master/install.py#L17) of the installer.

Yet the actual code is not set up as a WeeWX service. It's set up as a driver.

I don't see how the extension could have ever worked as written. You'll have to contact the extension author. 

Sorry.

**Comment by mjm455** [comment]
I tried installing it and running it as a simulator but that wasn't working either when I tried running the service.  But now I see your latest post here and you're saying that it needs to run as a driver.  Is that right?  I can run /etc/weewx/weewxd then in the background.  

I posted the error I had on the extension page but he was pointing out that I need to get the base install working first.  Maybe that is the issue with the ambient weather api is that it is looking to work as a service.  

I'm new to the weather station and running this software.  I just really want to keep my data local and have many years stored.  Thanks!
